### ISSUE TITLE
Sync mode is reset after an upgrade on windows

### ISSUE THREAD

**Issue by JanKalin** [paritytech/parity#4664]
I've noticed this behaviour yesterday, after installing Parity v1.5.3. Periodically the application abruptly loses connectivity. Typical lines in the log are:
```
2017-02-24 11:35:16      0/19/25 peers     57 MiB db    6 MiB chain  0 bytes queue  118 KiB sync
2017-02-24 11:35:19  Imported #3241589 a2f2â€¦e1aa (6 txs, 0.13 Mgas, 2.60 ms, 1.19 KiB)
2017-02-24 11:35:51      0/ 0/25 peers     57 MiB db    6 MiB chain  0 bytes queue  118 KiB sync
2017-02-24 11:36:26      0/ 0/25 peers     57 MiB db    6 MiB chain  0 bytes queue  118 KiB sync
.
.
2017-02-24 12:23:16      0/ 0/25 peers     57 MiB db    6 MiB chain  0 bytes queue  118 KiB sync
2017-02-24 12:23:51      0/ 0/25 peers     57 MiB db    6 MiB chain  0 bytes queue  118 KiB sync
2017-02-24 12:24:19  Imported #3241812 7740â€¦cc06 (2 txs, 0.09 Mgas, 9.72 ms, 0.76 KiB) + another 3 block(s) containing 21 tx(s)
2017-02-24 12:24:26  Imported #3241813 e3c3â€¦2220 (23 txs, 0.66 Mgas, 10.70 ms, 3.51 KiB)
2017-02-24 12:24:26      0/10/25 peers     57 MiB db    6 MiB chain  0 bytes queue  118 KiB sync
```
Sockets are left in a CLOSE_WAIT state 
![image](https://cloud.githubusercontent.com/assets/18305372/23302097/8f1ef2d0-fa8d-11e6-907e-096e8bc4884b.png)

This state usually resolves by itself. However, it can also be resolved by connecting from a browser!
![image](https://cloud.githubusercontent.com/assets/18305372/23302763/7b21c610-fa90-11e6-92d4-13432ae1b9a4.png)

Before connection from the browser the CPU and I/O are 0 and you can see the spike that occurs when connection is made.
![image](https://cloud.githubusercontent.com/assets/18305372/23302778/8e0b960c-fa90-11e6-9614-2949fb8674eb.png)

This was reproduced twice and recorded in the enclosed [parity_log.txt](https://github.com/ethcore/parity/files/799206/parity_log.txt) at `2017-02-24 12:24:19` and `2017-02-24 12:54:06`

I have just downloaded v1.5.4 and will report back if the behaviour perists or not.

The OS is a relatively fresh install of Windows 7 64bit, memory is not an issue - there is 1.5GB of 6GB free.

**Comment by arkpar** [comment]
Make sure that mode of operation under Parity settings in the UI is not set to "Parity syncs initially, then sleeps and wakes regularly to resync"

**Comment by JanKalin** [comment]
That was it. Although I'm 99.9% sure that I had set this to `Parity continuously syncs the chain`. Is it possible that the setting is somehow changed when Parity upgrades?

**Comment by arkpar** [comment]
Should not be the case. Did you use the installer to upgrade? What was the previous version?

**Comment by JanKalin** [comment]
```
2017-02-03 10:26:07  Starting Parity/v1.4.5-beta-a028d04-20161126/x86_64-windows-msvc/rustc1.13.0
2017-02-10 15:52:37  Starting Parity/v1.5.2-beta-948a538-20170208/x86_64-windows-msvc/rustc1.14.0
2017-02-23 13:48:40  Starting Parity/v1.5.3-beta-0f99b1e-20170220/x86_64-windows-msvc/rustc1.14.0
2017-02-23 14:42:41  Starting Parity/v1.5.2-beta-948a538-20170208/x86_64-windows-msvc/rustc1.14.0
2017-02-23 14:50:11  Starting Parity/v1.5.3-beta-0f99b1e-20170220/x86_64-windows-msvc/rustc1.14.0
2017-02-24 13:14:50  Starting Parity/v1.5.4-beta-74b850e-20170223/x86_64-windows-msvc/rustc1.14.0
```
And the "problem" first appeared at `2017-02-23 13:59:52`, around 11 minutes after I had first run v1.5.3. (the v1.5.2 sandwitched between v1.5.3 was due to investigation into https://github.com/ethcore/parity/issues/4646)

I used the `InstallParity.exe` linked from `https://github.com/ethcore/parity/releases/`. And, thinking back, I'm even more certain that I did not change the setting.

P.S., would it be possible to include the version number in the Windows installer?

**Comment by 5chdn** [comment]
Closing as _works as expected_. You can use the different operating modes to change the synchronization behaviour:

      --mode MODE                    Set the operating mode. MODE can be one of:
                                     last - Uses the last-used mode, active if none.
                                     active - Parity continuously syncs the chain.
                                     passive - Parity syncs initially, then sleeps and
                                     wakes regularly to resync.
                                     dark - Parity syncs only when the RPC is active.
                                     offline - Parity doesn't sync. (default: last).

Let me know if you have any other questions.

**Comment by JanKalin** [comment]
On my computer the mode of synchronisation *definitely* changes when upgrading Parity. This time I'm certain that I didn't change it during the installation. Procedure:
- Running Parity v1.5.4 in the `Parity continuously syncs the chain` mode
- Download Parity v1.5.7
- Kill Parity
- Run InstallParity.exe
- Stop PTray that gets launched after installation
- Close Parity tab in browser
- Run `detach "C:\Progra~1\Ethcore\Parity\parity.exe --log-file C:\Users\jank\.parity\parity.log"`
- Wait a bit
- Parity stops syncing
- Open Parity in browser, the mode is set to `Parity syncs initially, then sleeps and wakes regularly to resync`

Here's the log around the upgrade time:
[sample_log.txt](https://github.com/ethcore/parity/files/833455/sample_log.txt)

**Comment by 5chdn** [comment]
On Gitter someone reported his mode of operation was reset to 'offline' after switching from Installer version to Brew version on Mac.

**Comment by arkpar** [comment]
Fixed in #4970

**Comment by JanKalin** [comment]
Sorry, still here...
![clipboard01](https://cloud.githubusercontent.com/assets/18305372/25654053/ee78d63c-2fef-11e7-85d1-6ceaec35c8ac.png)
![clipboard02](https://cloud.githubusercontent.com/assets/18305372/25654057/f2373d5e-2fef-11e7-928e-82ce09e836f7.png)

**Comment by 5chdn** [comment]
This issue is labelled with _bug_: The client fails to follow expected behavior. It's assigned to @arkpar but not linked to a milestone.

Please, decide on a deadline within 7 days or close it if it's fixed already.

**Comment by JanKalin** [comment]
In v1.7 the setting is preserved:
![clipboard01](https://user-images.githubusercontent.com/18305372/28820245-35459408-76b1-11e7-9157-65f6bb0a4707.png)
![clipboard02](https://user-images.githubusercontent.com/18305372/28820250-38aef79c-76b1-11e7-81dc-10135ec2907b.png)

**Comment by 5chdn** [comment]
Thanks.

**Comment by JanKalin** [comment]
Oh, well, here it goes again. Updating to 1.7.3 changed the setting to "syncs initially, then sleeps,..."

**Comment by 5chdn** [comment]
:sob: :rofl:

**Comment by JanKalin** [comment]
Perhaps it's something specific on my machine or upgrade procedure. The  way I do the upgrade is to:
- Download InstallParity.exe
- Kill running Parity process
- Install new Parity
- Immediately after Parity starts use the ptray app to:
  - Uncheck "Start At Login"
  - Exit the app
- Then I run my own command `par.btm`, which contains lines line `detach "C:\Progra~1\Parity\Parity\parity.exe --log-file C:\Users\jank\.parity\parity.log"` and `exit`

**Comment by JanKalin** [comment]
In Parity//v1.8.1-beta-0e4a06d-20171020/x86_64-windows-msvc/rustc1.20.0 this didn't happen. However, this time I upgraded by downloading and copying the Parity.exe directly to the `C:\Program Files\Parity` directory, without running the installer.

**Comment by 5chdn** [comment]
Won't fix, we will remove the installers with 1.12.
